packages/firebase_ml/lib/src/model_management/firebase_model_manager.dart

```diff
@@ -18,32 +19,35 @@ import 'firebase_model_download_conditions.dart';
 /// https://firebase.google.com/docs/reference/android/com/google/
 /// firebase/ml/common/modeldownload/FirebaseModelManager
 class FirebaseModelManager {
-  static const MethodChannel _channel =
+  FirebaseModelManager._();
+
+  @visibleForTesting
+
+  /// Means for communication with native platform code
+  static const MethodChannel channel =
       MethodChannel('plugins.flutter.io/firebase_ml');
 
   /// Singleton of [FirebaseModelManager].
-  static final FirebaseModelManager instance = FirebaseModelManager();
+  static final FirebaseModelManager instance = FirebaseModelManager._();
 
   /// Initiates the download of remoteModel if the download hasn't begun.
   Future<void> download(FirebaseRemoteModel model,
       FirebaseModelDownloadConditions conditions) async {
-    // Example call
-    await _channel.invokeMethod(
-        "FirebaseModelManager#download", {'model': model.toString()});
+    Map modelMap = await channel.invokeMethod("FirebaseModelManager#download",
+        {'model': model.modelName, 'conditions': conditions.toMap()});
```

Comment:
we are only using `model.modelName`, should we just take in a `String` of modelName only?

That way we can make FirebaseRemoteModel immutable by making modelHash final, and we just return the new `FirebaseRemoteModel`?

Reply:
This was in the old version, doesn't exist after "Java restructuring" commit (5b4b0b2)